crates/matrix-sdk-sqlite/src/event_cache_store.rs

```diff
@@ -380,7 +380,14 @@ impl EventCacheStore for SqliteEventCacheStore {
 
         self.acquire()
             .await?
-            .with_transaction(move |txn| -> Result<_, Self::Error> {
+            .interact(move |conn| -> Result<_, Self::Error> {
+                // Start the transaction in IMMEDIATE mode since all updates may cause writes, to
+                // avoid read transactions upgrading to write mode and causing SQLITE_BUSY errors.
+                // See also: https://www.sqlite.org/lang_transaction.html#deferred_immediate_and_exclusive_transactions
+                conn.set_transaction_behavior(TransactionBehavior::Immediate);
```

Comment:
Does the thing go back to the default after we call `drop()` on the connection?

Reply:
I haven't seen any sign of such thing, but since most accesses are write anyways (even reading from the media cache causes a write access to update the last_touch time), I figured it was OK. That being said, it's not very future-proof, and there are also a few read transactions. So I think I can manually set it back to `Deferred` at the end.

Reply:
Yeah, let's set it back, otherwise me might slow down things or run into more database locked errors.